app/src/main/java/com/jaredrummler/mobiletest/ui/home/HomeFragment.kt

```diff
@@ -0,0 +1,47 @@
+package com.jaredrummler.mobiletest.ui.home
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.jaredrummler.mobiletest.R
+import com.jaredrummler.mobiletest.ui.BaseFragment
+import com.jaredrummler.mobiletest.ui.MainView
+import kotlinx.android.synthetic.main.fragment_home.fabQrCode
+import kotlinx.android.synthetic.main.fragment_home.fabScan
+import kotlinx.android.synthetic.main.fragment_home.floatingActions
+
+class HomeFragment : BaseFragment() {
+
+  override fun getActionBarTitle(): Int = R.string.home
+
+  override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+    return inflater.inflate(R.layout.fragment_home, container, false)
+  }
+
+  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+    super.onViewCreated(view, savedInstanceState)
+    fabQrCode.setOnClickListener {
+      floatingActions.collapse()
+      (activity as? MainView)?.showQrCode()
+    }
+    fabScan.setOnClickListener {
+      floatingActions.collapse()
+      (activity as? MainView)?.scanQrCode()
+    }
+  }
+
+  override fun handleOnBackPressed(): Boolean {
+    if (floatingActions.isExpanded) {
```

Comment:
You can throw `return` outside condition statement. 

```
{
   return if (floatingActions.isExpanded){
      ....
      true
   }
   false
}
```

Reply:
Nice suggestion. I do think this would need an `else false` at the end. I do wish the ternary operator existed in kotlin, although it wouldn't have been useful here.